api/src/main/java/org/openmrs/validator/PersonValidator.java

```diff
@@ -68,53 +68,55 @@ public void validate(Object target, Errors errors) {
 		}
 		
 		Person person = (Person) target;
-		
-		int index = 0;
-		boolean atLeastOneNonVoidPersonNameLeft = false;
-		for (PersonName personName : person.getNames()) {
-			errors.pushNestedPath("names[" + index + "]");
-			personNameValidator.validate(personName, errors);
-			if (!personName.getVoided()) {
-				atLeastOneNonVoidPersonNameLeft = true;
-			}
-			errors.popNestedPath();
-			index++;
-		}
-		if (!person.getVoided() && !atLeastOneNonVoidPersonNameLeft) {
+
+		validatePersonName(errors, person);
+
+		if (!person.getVoided() && person.hasZeroNotVoidedNames()) {
 			errors.rejectValue("names", "Person.shouldHaveAtleastOneNonVoidedName");
 		}
 		
-		// validate the personAddress
-		index = 0;
-		for (PersonAddress address : person.getAddresses()) {
-			try {
-				errors.pushNestedPath("addresses[" + index + "]");
-				ValidationUtils.invokeValidator(personAddressValidator, address, errors);
-			}
-			finally {
-				errors.popNestedPath();
-				index++;
-			}
-		}
-		
+		validatePersonAddress(errors, person);
 		validateBirthDate(errors, person.getBirthdate());
 		validateDeathDate(errors, person.getDeathDate(), person.getBirthdate());
 		
 		if (person.getVoided()) {
 			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "voidReason", "error.null");
 		}
+		
 		if (person.getDead()) {
-			if(person.getCauseOfDeath() != null && person.getCauseOfDeathNonCoded() != null) {
+			if (person.hasBothCauseOfDeathAndNonCodedCauseOfDeath()) {
```

Comment:
Are these new method replacements for making sonar happy?

Reply:
Yes, exactly. Logical operations like `&&` and `||` increase the code complexity.
But actually the other changes that I've made should be enough to fix the sonar issue and I'm starting to think that these two method extractions around causes of death are maybe a little bit too much so I'll revert them. This change doesn't make the code easier to read, maybe even make it harder 🙇